core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +286,2259 @@ def validate(cls, item):
+            'type': 'list',
+            'items': activity_references_dict_schema
+        }
+
+        return {
+            'activity_references': activity_references_schema
+        }
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        model_domain_object_instances = []
+
+        try:
+            for obj in item.activity_references:
+                model_domain_object_instances.append(
+                    activity_domain.ActivityReference(obj['type'], obj['id']))
+        except Exception as e:
+            cls.errors['fetch properties'] = (
+                'Model id %s: Model properties cannot be fetched completely '
+                'with the error %s') % (item.id, e)
+            return []
+
+        return model_domain_object_instances
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
 
     @classmethod
     def _get_external_id_relationships(cls, item):
+        exploration_ids = []
+        collection_ids = []
+
+        try:
+            for obj in item.activity_references:
+                if obj['type'] == constants.ACTIVITY_TYPE_EXPLORATION:
+                    exploration_ids.append(obj['id'])
+                elif obj['type'] == constants.ACTIVITY_TYPE_COLLECTION:
+                    collection_ids.append(obj['id'])
+        except Exception as e:
+            cls.errors['fetch properties'] = (
+                'Model id %s: Model properties cannot be fetched completely '
+                'with the error %s') % (item.id, e)
+            return {}
+
         return {
-            'activity_ids': (exp_models.ExplorationModel, item.activity_ids),
+            'exploration_ids': (exp_models.ExplorationModel, exploration_ids),
             'collection_ids': (
+                collection_models.CollectionModel, collection_ids),
+        }
+
+    @classmethod
+    def _get_validation_functions(cls):
+        return []
+
+
+class RoleQueryAuditModelValidator(BaseModelValidator):
+    """Class for validating RoleQueryAuditModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        regex_string = '%s\\.\\d*\\.%s\\.\\d*$' % (item.user_id, item.intent)
+        return regex_string
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        return []
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {'user_id': (user_models.UserSettingsModel, [item.user_id])}
+
+    @classmethod
+    def _validate_user_id_belongs_to_admin(cls, item):
+        """Validate that user id of model belongs to an admin.
+
+        Args:
+            item: RoleQueryAuditModel to validate.
+        """
+        _, user_id_models = (cls.external_models['user_id'])
+        user_id_model = user_id_models[0][1]
```

Comment:
Yes, I agree.

The way we fetch external models is as follows:

`cls.errors['field name'] = (model class, zip(ids, models))`

where field name is the field for which we are fetching models like exploration_ids.

For example,

`cls.errors['exploration_ids'] = (exp_models.ExplorationModel, [(id1, entity1), (id2, entity2)...]`

In this case we will have only one user model since we are fetching model for only one id. So,

```
(user_models.UserSettingsModel, [(id1, entity1)] = cls.external_models['user_id']
user_id_models = [(id1, entity1)]
user_id_model = entity1 = user_id_models[0][1]
```

This makes it look unclear. So, now I am iterating over the models in a loop like [here](https://github.com/oppia/oppia/pull/6857/files#diff-ecafbbaeb117bcae187e78e5b33e120fR586). Is that fine?